hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordReader.java

```diff
@@ -198,41 +190,42 @@ protected AbstractHoodieLogRecordReader(FileSystem fs, String basePath, List<Str
     this.path = basePath;
     this.useScanV2 = useScanV2;
 
-    // Key fields when populate meta fields is disabled (that is, virtual keys enabled)
-    if (!tableConfig.populateMetaFields()) {
-      this.populateMetaFields = false;
-      this.simpleKeyGenFields = Option.of(
-          Pair.of(tableConfig.getRecordKeyFieldProp(), tableConfig.getPartitionFieldProp()));
-    }
-    this.partitionName = partitionName;
-    this.recordType = recordMerger.getRecordType();
-  }
+    if (keyFieldOverride.isPresent()) {
+      // TODO elaborate
+      checkState(partitionNameOverride.isPresent());
 
-  protected String getKeyField() {
-    if (this.populateMetaFields) {
-      return HoodieRecord.RECORD_KEY_METADATA_FIELD;
+      this.populateMetaFields = false;
+      this.recordKeyField = keyFieldOverride.get();
+      this.partitionPathFieldOpt = Option.empty();
+    } else if (tableConfig.populateMetaFields()) {
+      this.populateMetaFields = true;
+      this.recordKeyField = HoodieRecord.RECORD_KEY_METADATA_FIELD;
+      this.partitionPathFieldOpt = Option.of(HoodieRecord.PARTITION_PATH_METADATA_FIELD);
+    } else {
+      this.populateMetaFields = false;
```

Comment:
can you add java docs for all 3 cases what each one is meant for.
might be confusing since we have virtual keys for data table, virtual keys for metadata table, keyField override is just for metadata table and so could get confusing.

Reply:
Added java-doc for the first branch, let me know if that makes the remaining 2 clearer